vumi/persist/fields.py

```diff
@@ -56,24 +57,50 @@ def _add_index(self, modelobj, value):
             return
         modelobj._riak_object.add_index(self.index_name, str(value))
 
+    def get_riak_data(self, modelobj, default=None, key=None):
+        if key is None:
+            key = self.key
+        return modelobj._riak_object.get_data().get(key, default)
+
+    def set_riak_data(self, modelobj, raw_value, key=None):
+        if key is None:
+            key = self.key
+        old_raw_value = modelobj._riak_object.get_data().get(key)
+        modelobj._riak_object.set_data_field(key, raw_value)
```

Comment:
Good point. I'll change that.

(This code originally lived at a higher level of abstraction, so we didn't actually know if the data would be modified by the write.)